### PR TITLE
Worldpay IP verification

### PR DESCRIPTION
**Pull request by andrewpaliga**
Adding IP range verification to Worldpay Offsite

@girasquid @j-mutter 

@louiskearns

### CODE REVIEW

**Comment by andrewpaliga** [comment]
ping

File: lib/offsite_payments/integrations/world_pay.rb [hunk 1]
```diff
@@ -230,6 +230,12 @@ def custom_params
           return @custom_params ||= read_custom_params
         end
 
+        # Check if the request comes from IP range 195.35.90.0 – 195.35.91.255
+        def valid_sender?(ip)
+          return true if OffsitePayments.mode == :test 
+          !!(/^195\.35\.(9[0-1])\.([0-9]|[1-9][0-9]|1([0-9][0-9])|2([0-4][0-9]|5[0-5]))$/.match(ip))
```
Comment (girasquid) [inline lib/offsite_payments/integrations/world_pay.rb]: This is a hairy regex. Could you instead use `IPAddr#include?`?: http://ruby-doc.org/stdlib-2.0/libdoc/ipaddr/rdoc/IPAddr.html#method-i-include-3F

Comment (j-mutter) [inline lib/offsite_payments/integrations/world_pay.rb]: :+1: to that. I think you want `IPAddr.new("195.35.90.0/23")` if my subnet mask knowledge is still up to snuff

**Comment by andrewpaliga** [comment]
looks good now?

**Comment by girasquid** [comment]
:+1:

File: lib/offsite_payments/integrations/world_pay.rb [hunk 2]
```diff
@@ -230,6 +231,12 @@ def custom_params
           return @custom_params ||= read_custom_params
         end
 
+        # Check if the request comes from IP range 195.35.90.0 – 195.35.91.255
+        def valid_sender?(ip)
+          return true if OffsitePayments.mode == :test 
+          IPAddr.new("195.35.90.0/23").include?(IPAddr.new(ip))
```
Comment (j-mutter) [inline lib/offsite_payments/integrations/world_pay.rb]: These guys really have *that* many IPs? I know this works, but I wish we didn't have to allow 500+ addresses

File: test/unit/integrations/world_pay/world_pay_notification_test.rb [hunk 3]
```diff
@@ -65,6 +65,14 @@ def test_custom_parameters
     assert_equal 'Custom Value 3', notification.custom_params[:custom_3]
   end
 
+  def test_valid_sender
+    OffsitePayments.mode = :production 
+    assert @world_pay.valid_sender?('195.35.90.0')
+    assert @world_pay.valid_sender?('195.35.90.11')
+    assert @world_pay.valid_sender?('195.35.91.255')
+    assert !@world_pay.valid_sender?('195.35.89.255')
```
Comment (j-mutter) [inline test/unit/integrations/world_pay/world_pay_notification_test.rb]: `refute` > `assert !` :innocent:

File: lib/offsite_payments/integrations/world_pay.rb (hunk 2 above)
Comment (andrewpaliga) [inline lib/offsite_payments/integrations/world_pay.rb]: That's the range they gave us.

**Pull request merged**